bin/src/hickory-dns.rs

```diff
@@ -171,6 +174,41 @@ struct Cli {
     #[cfg(feature = "prometheus-metrics")]
     #[clap(long = "disable-prometheus", conflicts_with = "prometheus_listen_addr")]
     disable_prometheus: bool,
+
+    /// Name server identifier (NSID) payload for EDNS responses.
+    /// Use `0x` prefix for hex-encoded data. Mutually exclusive with --nsid-hostname
+    #[clap(long = "nsid", value_name = "NSID", group = "nsid_group")]
+    nsid: Option<NSIDPayloadValue>,
+
+    /// Use the system hostname as the name server identifier (NSID) payload
+    /// for EDNS responses.
+    /// Mutually exclusive with --nsid
+    #[clap(long = "nsid-hostname", group = "nsid_group")]
+    nsid_hostname: bool,
+}
+
+/// A wrapper type to allow both string, and hex-encoded NSID values
+///
+/// We don't want to take a `hex` dependency in the hickory-proto crate where the NSIDPayload
+/// type is defined.
+#[derive(Debug, Clone)]
+struct NSIDPayloadValue(NSIDPayload);
```

Comment:
This feels like a little cumbersome. Suggest we use the `value_parser` argument instead on the `nsid` option and only define the parser function.

Reply:
Ah! Much nicer, thank you. Fixed.